c++/src/kj/compat/http.c++

```diff
@@ -5380,6 +6085,43 @@ private:
     return kj::heap<BrokenWebSocket>(KJ_EXCEPTION(FAILED,
         "received bad WebSocket handshake", errorMessage));
   }
+
+  kj::Own<kj::AsyncIoStream> accept(uint statusCode, kj::StringPtr statusText,
+                                    const HttpHeaders& headers) override {
+    auto method = KJ_REQUIRE_NONNULL(currentMethod, "already called send()");
+    currentMethod = nullptr;
+    KJ_REQUIRE(method.is<HttpConnectMethod>(), "Only use accept() with CONNECT requests.");
+
+    KJ_REQUIRE(statusCode >= 200 && statusCode < 300, "the statusCode must be 2xx for accept.");
+    tunnelRejected = nullptr;
+
+    httpOutput.writeHeaders(headers.serializeResponse(statusCode, statusText));
+
+    auto releasedBuffer = httpInput.releaseBuffer();
+    return heap<ConnectPipeEnd>(
+        kj::Own<kj::AsyncIoStream>(&stream, kj::NullDisposer::instance),
```

Comment:
Do we want to do something like the `deferNoteClosed` mechanism that we do in `acceptWebSocket()`?